app/src/main/java/org/p2p/wallet/newsend/smartselection/SmartSelectionCoordinator.kt

```diff
@@ -0,0 +1,142 @@
+package org.p2p.wallet.newsend.smartselection
+
+import timber.log.Timber
+import java.util.concurrent.CancellationException
+import java.util.concurrent.Executors
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import org.p2p.core.token.Token
+import org.p2p.wallet.feerelayer.model.FeeRelayerFee
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.newsend.model.FeePayerState
+import org.p2p.wallet.newsend.model.smartselection.FeePayerFailureReason
+import org.p2p.wallet.newsend.model.smartselection.SmartSelectionState
+import org.p2p.wallet.newsend.smartselection.handler.TriggerHandler
+import org.p2p.wallet.newsend.smartselection.strategy.StrategyExecutor
+
+private const val TAG = "SmartSelectionCoordinator"
+
+class SmartSelectionCoordinator(
+    private val strategyExecutor: StrategyExecutor,
+    private val triggerHandlers: List<TriggerHandler>,
+    dispatchers: CoroutineDispatchers
+) : CoroutineScope {
+
+    override val coroutineContext: CoroutineContext = SupervisorJob() + dispatchers.io +
+        Executors.newSingleThreadExecutor().asCoroutineDispatcher()
```

Comment:
`Executors.newSingleThreadExecutor().asCoroutineDispatcher()` overrides `dispatchers.io`. 
As a result, all launches are launched in a single thread dispatcher, so `dispatchers.io` is unnecessary in this context. 
I suggest to add a function or variable into the CoroutineDispatchers that can create a new threaded dispatcher. This way, we can change its behavior for testing and manage this dispatcher from koin.